Unit test failure on CI system

https://ci.vmware.run/vmware/vic/4258
```
Failed to run date for comparison data: wait: no child processestime="2016-08-23T03:35:46Z" level=info msg="Reaped zombie process PID 20864\n" 
time="2016-08-23T03:35:46Z" level=info msg="pathlookup exit code: 0" 
SIGQUIT: quit
PC=0x461263 m=2

goroutine 0 [idle]:
runtime.futex(0x144de38, 0x0, 0x7fb900ca6d88, 0x0, 0x7fb900000000, 0x460f86, 0x3c, 0x0, 0x7fb900ca6dd0, 0x41076b, ...)
	/usr/local/go/src/runtime/sys_linux_amd64.s:388 +0x23
runtime.futexsleep(0x144de38, 0x0, 0xdf8475800)
	/usr/local/go/src/runtime/os_linux.go:62 +0xd7
runtime.notetsleep_internal(0x144de38, 0xdf8475800, 0x1)
	/usr/local/go/src/runtime/lock_futex.go:174 +0xeb
runtime.notetsleep(0x144de38, 0xdf8475800, 0xa43755622a2c9)
	/usr/local/go/src/runtime/lock_futex.go:194 +0x56
runtime.sysmon()
	/usr/local/go/src/runtime/proc.go:3618 +0x14c
runtime.mstart1()
	/usr/local/go/src/runtime/proc.go:1126 +0x11e
runtime.mstart()
	/usr/local/go/src/runtime/proc.go:1096 +0x64

goroutine 1 [chan receive, 9 minutes]:
testing.(*T).Run(0xc4201f00c0, 0xca321f, 0xb, 0xcfd7f0, 0xc420049c78)
	/usr/local/go/src/testing/testing.go:647 +0x316
testing.RunTests.func1(0xc4201f00c0)
	/usr/local/go/src/testing/testing.go:793 +0x6d
testing.tRunner(0xc4201f00c0, 0xc420034d78)
	/usr/local/go/src/testing/testing.go:610 +0x81
testing.RunTests(0xcfda80, 0x1425d80, 0x1, 0x1, 0xc4200001a0)
	/usr/local/go/src/testing/testing.go:799 +0x2f5
testing.(*M).Run(0xc420034ef8, 0x1000000077)
	/usr/local/go/src/testing/testing.go:743 +0x85
github.com/vmware/vic/cmd/vic-init.TestMain(0xc420034ef8)
	/drone/src/github.com/vmware/vic/cmd/vic-init/tether_test.go:168 +0x4c
main.main()
	github.com/vmware/vic/cmd/vic-init/_test/_testmain.go:100 +0x1bd

goroutine 17 [syscall, 9 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 5 [syscall, 9 minutes]:
os/signal.signal_recv(0x1436fe0)
	/usr/local/go/src/runtime/sigqueue.go:116 +0x157
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 18 [chan receive, 9 minutes]:
github.com/vmware/vic/cmd/vic-init.testTeardown(0xc4201f0180)
	/drone/src/github.com/vmware/vic/cmd/vic-init/tether_test.go:242 +0x62
github.com/vmware/vic/cmd/vic-init.TestRestart(0xc4201f0180)
	/drone/src/github.com/vmware/vic/cmd/vic-init/restart_test.go:98 +0xd0a
testing.tRunner(0xc4201f0180, 0xcfd7f0)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec

goroutine 19 [chan receive, 9 minutes]:
github.com/vmware/vic/lib/tether.(*tether).Start(0xc4201189c0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/lib/tether/tether.go:170 +0x1e3
github.com/vmware/vic/cmd/vic-init.StartTether.func1(0xc4201f0180)
	/drone/src/github.com/vmware/vic/cmd/vic-init/tether_test.go:186 +0x39
created by github.com/vmware/vic/cmd/vic-init.StartTether
	/drone/src/github.com/vmware/vic/cmd/vic-init/tether_test.go:190 +0x256

goroutine 20 [select, 9 minutes, locked to thread]:
runtime.gopark(0xcfe2f8, 0x0, 0xc9f3d4, 0x6, 0x18, 0x2)
	/usr/local/go/src/runtime/proc.go:259 +0x13a
runtime.selectgoImpl(0xc420020730, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:423 +0x11d9
runtime.selectgo(0xc420020730)
	/usr/local/go/src/runtime/select.go:238 +0x1c
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:304 +0x2f3
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 21 [chan send, 9 minutes]:
github.com/vmware/vic/cmd/vic-init.(*Mocker).HandleSessionExit.func1()
	/drone/src/github.com/vmware/vic/cmd/vic-init/tether_test.go:117 +0x49
github.com/vmware/vic/lib/tether.(*tether).handleSessionExit(0xc4201189c0, 0xc42012e480)
	/drone/src/github.com/vmware/vic/lib/tether/tether.go:324 +0x4bc
github.com/vmware/vic/lib/tether.(*tether).childReaper.func1.1(0xc42003af7c, 0xc4201189c0)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:96 +0x2e9
github.com/vmware/vic/lib/tether.(*tether).childReaper.func1(0xc4201189c0)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:106 +0x9b
created by github.com/vmware/vic/lib/tether.(*tether).childReaper
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:108 +0x1e3

goroutine 22 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fb9013eb108, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420016140, 0x72, 0xc420039d08, 0xc4200140b8)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420016140, 0x14347e0, 0xc4200140b8)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).accept(0xc4200160e0, 0x0, 0x14331e0, 0xc4200c50c0)
	/usr/local/go/src/net/fd_unix.go:419 +0x238
net.(*TCPListener).accept(0xc420028030, 0xc420011d40, 0xc420039e18, 0x8b7ab0)
	/usr/local/go/src/net/tcpsock_posix.go:132 +0x2e
net.(*TCPListener).AcceptTCP(0xc420028030, 0xc4200803dc, 0xc420039e08, 0x5c2850)
	/usr/local/go/src/net/tcpsock.go:209 +0x49
net/http.tcpKeepAliveListener.Accept(0xc420028030, 0xc420011d10, 0xaf33e0, 0x1425030, 0xbe6080)
	/usr/local/go/src/net/http/server.go:2608 +0x2f
net/http.(*Server).Serve(0xc420080380, 0x1437da0, 0xc420028030, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2273 +0x1ce
net/http.(*Server).ListenAndServe(0xc420080380, 0xc420080380, 0x0)
	/usr/local/go/src/net/http/server.go:2219 +0xb4
net/http.ListenAndServe(0xca3851, 0xc, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2351 +0xa0
github.com/vmware/vic/cmd/vic-init.(*Mocker).Start.func1()
	/drone/src/github.com/vmware/vic/cmd/vic-init/tether_test.go:74 +0x4c
created by github.com/vmware/vic/cmd/vic-init.(*Mocker).Start
	/drone/src/github.com/vmware/vic/cmd/vic-init/tether_test.go:75 +0x35

rax    0xfffffffffffffffc
rbx    0x0
rcx    0x461263
rdx    0x0
rdi    0x144de38
rsi    0x0
rbp    0x7fb900ca6d98
rsp    0x7fb900ca6d50
r8     0x0
r9     0x0
r10    0x7fb900ca6d88
r11    0x246
r12    0x0
r13    0xf3
r14    0x30
r15    0x3
rip    0x461263
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
*** Test killed with quit: ran too long (10m0s).
FAIL	github.com/vmware/vic/cmd/vic-init	600.005s
make: *** [test-job-github.com/vmware/vic/cmd] Error 1
Makefile:227: recipe for target 'test-job-github.com/vmware/vic/cmd' failed
[info] build failed (exit code 2)
```

--- a/cmd/vic-init/restart_test.go
+++ b/cmd/vic-init/restart_test.go
@@ -90,6 +90,10 @@ func TestRestart(t *testing.T) {
 	// read the output from the session
 	log := Mocked.SessionLogBuffer.Bytes()
 
+	// the tether has to be stopped before comparison on the reaper may swaller exec.Wait
+	tthr.Stop()
+	<-Mocked.Cleaned
+
 	// run the command directly
 	out, err := exec.Command("/bin/date", "--reference=/").Output()
 	if err != nil {

--- a/lib/tether/cmd_test.go
+++ b/lib/tether/cmd_test.go
@@ -147,6 +147,9 @@ func TestAbsPath(t *testing.T) {
 	// read the output from the session
 	log := mocker.SessionLogBuffer.Bytes()
 
+	// block until tether exits
+	<-mocker.Cleaned
+
 	// run the command directly
 	out, err := exec.Command("/bin/date", "--reference=/").Output()
 	if err != nil {

